inst/stan/functions/rt.stan

```diff
@@ -1,72 +1,36 @@
-// Update an individual breakpoint effect
-real update_breakpoints(real input_R, real[] bp_effects,
-                        int bp_index, int at_bp,
-                        int stationary) {
-  real R = input_R;
-  if (stationary) {
-    if (bp_index > 0) {
-      R += sum(bp_effects[1:bp_index]);
-      }
-  }else{
-    if (at_bp) {
-      R += bp_effects[bp_index];
-    }
-  }               
-  return(R);
-}
-// update an individual R
-real update_R(vector R, vector noise, int noise_terms,
-              int index, int stationary) {
-  real cR = R[index];
-  if (noise_terms > 0) {
-    if (stationary){
-      if (index <= noise_terms) {
-        cR += noise[index];
-      }else{
-        if (index > 1) {
-           cR = R[index - 1];
-        }
-      }
-    }else{
-      if (index <= (noise_terms + 1)) {
-        cR = R[index - 1] + noise[index - 1];
-      }else{
-        cR = R[index - 1];
-      }
-    }
-  }
-  return(cR);
-}
 // update a vector of Rts
 vector update_Rt(vector input_R, real log_R, vector noise, int[] bps,
                  real[] bp_effects, int stationary) {
   // define control parameters
-  int noise_terms = num_elements(noise);
-  int i_stationary = noise_terms > 0 ? stationary : 1;
   int t = num_elements(input_R);
   int bp_n = num_elements(bp_effects);
-  int bp_in = 0;
-  int at_bp = 0;
-  int index;
+  int bp_c = 0;
+  int gp_n = num_elements(noise);
+  // define result vectors
+  vector[t] bp = rep_vector(0, t);
+  vector[t] gp = rep_vector(0, t);
   vector[t] R;
-  // initialise Rt
-  if (i_stationary) {
-    R = rep_vector(log_R, t);
-    index = 1;
-  }else{
-    R[1] = log_R;
-    index = 2;
+  // initialise breakpoints
+  if (bp_n > 0) {
+    for (s in 1:t) {
+      if (bps[s]) {
+        bp_c += bps[s];
+        bp[s] = bp_effects[bp_c];
+      }
+    }
+    bp = cumulative_sum(bp);
   }
-  // iteratively update Rt
-  for (s in index:t) {
-    R[s] = update_R(R, noise, noise_terms, s, i_stationary);
-    if (bp_n > 0) {
-      at_bp = bps[s];
-      bp_in += at_bp;
-      R[s] = update_breakpoints(R[s], bp_effects, bp_in, at_bp, i_stationary);
+  //initialise gaussian process
+  if (gp_n) {
```

Comment:
suggest to settle on either `if (x)` or `if (x > 0)` and stay consistent.

Reply:
yes good point - should really just be `if (x)` everywhere.